### PR TITLE
fix: set time format right

### PR DESCRIPTION
**Pull request by jiuker**
## Description

fix: set time format right

https://github.com/minio/minio/blob/be45ffd8a432b594ecb9a8733168aa6986212218/cmd/xl-storage-format-v2.go#L1298-L1307
I see the metadata use `time.RFC3339Nano`.
If not , cause some parse error maybe. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/erasure-metadata.go [hunk 1]
```diff
@@ -626,10 +626,10 @@ func getInternalReplicationState(m map[string]string) ReplicationState {
 	for k, v := range m {
 		switch {
 		case equals(k, ReservedMetadataPrefixLower+ReplicationTimestamp):
-			tm, _ := time.Parse(http.TimeFormat, v)
+			tm, _ := time.Parse(time.RFC3339Nano, v)
```
Comment (klauspost) [inline cmd/erasure-metadata.go]: @poornas For backcompat - should we try both if one fails?

**Comment by harshavardhana** [comment]
Go parses this fine what is bug you are fixing @jiuker ?

File: cmd/erasure-metadata.go (hunk 1 above)
Comment (poornas) [reply cmd/erasure-metadata.go]: yes, that would be better

**Comment by poornas** [comment]
there shouldn't be any parse issues - these timestamps are only used for some metadata overwrite decisions, and we don't really care about nano precision for that.

**Review by poornas — APPROVED** [review]

File: cmd/erasure-metadata.go [hunk 2]
```diff
@@ -626,10 +626,16 @@ func getInternalReplicationState(m map[string]string) ReplicationState {
...
+			if tm.IsZero() {
+				tm, _ = time.Parse(http.TimeFormat, v)
+			}
```
Comment (harshavardhana) [inline cmd/erasure-metadata.go]: move this to `internal/amztime` package and call this `ParseReplicationTS(str string)` and parse using two different values return which ever succeeds.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**